lib/semantics/resolve-names.cc

```diff
@@ -2086,14 +2090,17 @@ void ModuleVisitor::BeginModule(const parser::Name &name, bool isSubmodule) {
 // If an error occurs, report it and return nullptr.
 Scope *ModuleVisitor::FindModule(const parser::Name &name, Scope *ancestor) {
   ModFileReader reader{context()};
-  auto *scope{reader.Read(name.source, ancestor)};
+  Scope *scope{reader.Read(name.source, ancestor)};
   if (!scope) {
     return nullptr;
   }
   if (scope->kind() != Scope::Kind::Module) {
     Say(name, "'%s' is not a module"_err_en_US);
     return nullptr;
   }
+  if (DoesScopeContain(scope, currScope())) {
+    Say(name, "Module '%s' cannot USE itself."_err_en_US);
```

Comment:
Add a C1406 comment here?

Reply:
I don't think that C1406 applies.  I'll reference the relevant text.

Reply:
R1409 use-stmt is USE ...

C1406 (R1409) A scoping unit shall not directly reference an intrinsic module and a nonintrinsic module of the same name.

Seems like this is what the message is checking.  I don't get any valid hits on C1406 in f18 code.

Reply:
That constraint prevents
```
USE,INTRINSIC::IEEE_ARITHMETIC
USE,NON_INTRINSIC::IEEE_ARITHMETIC`
```

Reply:
I see; the conjunction _and_ vs. _or_ makes the difference from what I was reading.  Sorry for the extra cycles.